src/pint/models/noise_model.py

```diff
@@ -378,18 +397,23 @@ def ecorr_basis_weight_pair(self, toas):
         The weights used are the square of the ECORR values.
 
         """
+        no_toa_params = self.validate_toas(toas, suppress_warning=True)
         tbl = toas.table
         t = (tbl["tdbld"].quantity * u.day).to(u.s).value
         ecorrs = self.get_ecorrs()
         umats = []
         for ec in ecorrs:
+            if ec.name in no_toa_params:
+                continue
             mask = ec.select_toa_mask(toas)
```

Comment:
Can't you just skip the ones that don't select any TOAs? And record them here if you need a list of the ones to skip, later?

Reply:
OK.